README.md

```diff
@@ -435,6 +355,399 @@ One App can be started via docker or if built from source by running `node lib/s
+* Using SSR Fetch Client with [`loadModuleData`](#loadModuleData)
+
+##### `validateStateConfig`
+**Module Type**
+* 🚫 Root Module 
+* ✅ Child Module 
+
+**Shape**
+```js
+if (!global.BROWSER) {
+  Module.appConfig = {
+    validateStateConfig: {
+      server: {
+        [settingName]: {
+          validate(settingValue) {
+            // Throw an error or return undefined
+          },
+        },
+      },
+      client: {
+        [settingName]: {
+          validate(settingValue) {
+            // Throw an error or return undefined
+          },
+        },
+      },
+    },
+  };
+}
+```
+
+The `validateStateConfig` allows a Child Module to validate settings passed from `provideStateConfig`. Each `settingName` object accepts a `validate(settingValue)` method. The `validate` function may throw an `Error` or return `undefined` depending on validity of the value supplied to the Module on load.
+
+If an `Error` is thrown, the Server will fail to startup or if already running will prevent [Holocron](https://github.com/americanexpress/holocron) from loading the Module dynamically.
+
+**📘 More Information**
+* [`provideStateConfig`](#providestateconfig)
+* Example: [Picky Frank's `appConfig`](./prod-sample/sample-modules/picky-frank/0.0.0/src/components/PickyFrank.jsx)
+
+##### `requiredSafeRequestRestrictedAttributes`
+**Module Type**
+* 🚫 Root Module 
+* ✅ Child Module 
+
+> 👮**Security Feature**: Limits headers and cookies from being passed to Redux's initial state.
+
+**Shape**
+```js
+if (!global.BROWSER) {
+  Module.appConfig = {
+    requiredSafeRequestRestrictedAttributes: {
+      headers: [String],
+      cookies: [String],
+    },
+  };
+}
+```
+
+The `requiredSafeRequestRestrictedAttributes` allows a Child Module to validate settings passed from `extendSafeRequestRestrictedAttributes`. Each whitelisted header in `headers` array and cookie in `cookies` array will be checked against the Root Module's `extendSafeRequestRestrictedAttributes` on the loading of the Child Module. If this does not match entries previously made in Root Module's `extendSafeRequestRestrictedAttributes`, the Child Module will fail to load.
+
+If an `Error` is thrown due to missing required cookies or headers, the Server will either 1) fail to startup or 2) if already running will prevent [Holocron](https://github.com/americanexpress/holocron) from loading the Module dynamically.
+
+**📘 More Information**
+* [`extendSafeRequestRestrictedAttributes`](#extendsaferequestrestrictedattributes)
+* Example: [Vitruvius Franklin's `appConfig`](./prod-sample/sample-modules/vitruvius-franklin/0.0.0/src/components/VitruviusFranklin.jsx)
+
+##### `appCompatibility`
+**Module Type**
+* ✅ Root Module 
+* ✅ Child Module  
+
+**Shape**
+```js
+if (!global.BROWSER) {
+  Module.appConfig = {
+    appCompatibility: String,
+  };
+}
+```
+
+The `appCompatibility` directive accepts a valid Semantic Version string specifying compatibility with specific One App versions.
```

Comment:
Can we link to the semantic versioning docs?

Reply:
Fixed in d8db2b7